jazz_events/index.js

```diff
@@ -40,382 +40,449 @@ module.exports.handler = (event, context, cb) => {
+		if (!event.body) {
+			reject({
+				result: "inputError",
+				message: "Service inputs not defined!"
+			});
+		}
+		if (!event.body.service_context) {
+			reject({
+				result: "inputError",
+				message: "service_context not provided!"
+			});
+		}
+		if (!event.body.event_handler) {
+			reject({
+				result: "inputError",
+				message: "event_handler not provided!"
+			});
+		}
+		if (!event.body.event_name) {
+			reject({
+				result: "inputError",
+				message: "event_name not provided!"
+			});
+		}
+		if (!event.body.service_name) {
+			reject({
+				result: "inputError",
+				message: "service_name not provided!"
+			});
+		}
+		if (!event.body.event_status) {
+			reject({
+				result: "inputError",
+				message: "event_status not provided!"
+			});
+		}
+		if (!event.body.event_type) {
+			reject({
+				result: "inputError",
+				message: "event_type not provided!"
+			});
+		}
+		if (!event.body.username) {
+			reject({
+				result: "inputError",
+				message: "username not provided!"
+			});
+		}
+		if (!event.body.event_timestamp) {
+			reject({
+				result: "inputError",
+				message: "event_timestamp not provided!"
+			});
+		}
+		resolve();
+	});
+}
 
-						Object.keys(event.body).forEach(function (key) {
-							if (key === "service_context") {
-								event_params.Item.SERVICE_CONTEXT = {
-									S: JSON.stringify(event.body.service_context)
-								};
-							} else if (key === "event_handler") {
-								event_params.Item.EVENT_HANDLER = {
-									S: event.body.event_handler
-								};
-							} else if (key === "event_name") {
-								event_params.Item.EVENT_NAME = {
-									S: event.body.event_name
-								};
-							} else if (key === "service_name") {
-								event_params.Item.SERVICE_NAME = {
-									S: event.body.service_name
-								};
-							} else if (key === "event_status") {
-								event_params.Item.EVENT_STATUS = {
-									S: event.body.event_status
-								};
-							} else if (key === "event_type") {
-								event_params.Item.EVENT_TYPE = {
-									S: event.body.event_type
-								};
-							} else if (key === "username") {
-								event_params.Item.USERNAME = {
-									S: event.body.username
-								};
-							} else if (key === "event_timestamp") {
-								event_params.Item.EVENT_TIMESTAMP = {
-									S: event.body.event_timestamp
-								};
-							} else {
-								if (event.body[key] === null) {
-									event_params.Item[key.toUpperCase()] = {
-										NULL: true
-									};
-								} else {
-									event_params.Item[key.toUpperCase()] = {
-										S: event.body[key]
-									};
-								}
+function validateEventInput(config, event, dynamodb){
+	logger.info("Inside validateEventInput:")
+	return new Promise((resolve, reject) => {
+		validateEventType(config, event, dynamodb)
+		.then(() => validateEventName(config, event, dynamodb))
+		.then(() => validateEventHandler(config, event, dynamodb))
+		.then(() => validateEventStatus(config, event, dynamodb))
```

Comment:
Please write one method as configurable instead of 3 methods.

Reply:
Are you suggesting using Promiser.all?

Reply:
Updated code using Promise.all